### PR TITLE
PR: Extend Demo

### PR DESCRIPTION
**Pull request by nelsonic**
The extended demo is working fine on `localhost`:
gitea-demo-homepage-updated

 https://gitea-server.fly.dev/demo-org/hello-world
gitea-repo-updaetd

But deploying it to Fly.io is proving more challenging than I had hoped. 
Opened a dedicated issue for the `ssh` key setup: https://github.com/dwyl/gitea-demo/issues/4 on Fly.io

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,77 +1,110 @@
...
+RUN apt-get update -y && apt-get install -y build-essential libstdc++6 openssl libncurses5 git \
+  && apt-get clean && rm -f /var/lib/apt/lists/*_*
 
-WORKDIR /app
+# Create dir for ssh key
+RUN mkdir -p /home/nobody/.ssh/
+# copy the keys you've created on your machine
+COPY keys/id_ed25519 /home/nobody/.ssh/                                               
+# add know host gitea server
+RUN ssh-keyscan -H gitea-server.fly.dev > /home/nobody/.ssh/known_hosts
```
Comment (nelsonic) [inline Dockerfile]: @SimonLab the `RUN ssh-keyscan -H gitea-server.fly.dev > /home/nobody/.ssh/known_hosts ` is already present in the `Dockerfile`... https://github.com/dwyl/gitea-demo/issues/4#issuecomment-1132669902

**Comment by nelsonic** [comment]
The reason I haven't assigned this PR for review yet is that there's a failing test:
https://github.com/dwyl/gitea-demo/runs/6581166946?check_suite_focus=true

image

I think we're going to have to add additional Mock/Test functions for this to pass on GitHub CI ... see: https://github.com/dwyl/gitea/issues/28

Going to take a look at it this evening after "work work".

**Comment by nelsonic** [comment]
Boom! Passing!! 💥

**Comment by nelsonic** [comment]
When reviewing this PR, visit the Fly.io App: https://gitea-demo.fly.dev/
phoenix-gitea-demo-on-fly io

And the `gitea` _server_ repo:
https://gitea-server.fly.dev/demo-org/hello-world
gitea-demo-file-updated

**Comment by nelsonic** [comment]
@SimonLab I've done a quick (self) proof-read of https://github.com/dwyl/gitea-demo/blob/extend-demo/part2.md and made a few updates. I feel this PR is now in a reviewable state. Thanks! 🙏

**Comment by nelsonic** [comment]
Very keen to test the CI/CD (specifically deploying the `ssh` key to Fly.io) so that we can _use_ it in the project. 🤞

**Review by SimonLab — APPROVED** [review]
LGTM

**Comment by nelsonic** [comment]
Follow-up deployment debugging required: #16

**Pull request merged**